source/simulator/initial_conditions.cc

```diff
@@ -342,12 +342,18 @@ namespace aspect
         // wants a function that represents all components of the
         // solution vector, so create such a function object
         // that is simply zero for all velocity components
+        auto lambda = [&](const Point<dim> &p) -> double
+        {
+          return adiabatic_conditions->pressure(p);
+        };
+
+        VectorFunctionFromScalarFunctionObject<dim> vector_function_object(
+          lambda,
+          pressure_comp,
+          introspection.n_components);
+
         VectorTools::interpolate (*mapping, dof_handler,
-                                  VectorFunctionFromScalarFunctionObject<dim> (std::bind (&AdiabaticConditions::Interface<dim>::pressure,
-                                                                               std::cref (*adiabatic_conditions),
-                                                                               std::placeholders::_1),
-                                                                               pressure_comp,
-                                                                               introspection.n_components),
+                                  vector_function_object,
```

Comment:
hmm, `pressure_function_object` feels easier to understand. I was always confused about this whole `VectorFunctionFromScalarFunctionObject` name (it only described what is technically necessary, but does not explain what it actually does, namely returning the pressure).